src/zcash/Proof.hpp

```diff
@@ -235,6 +235,42 @@ class ZCProof {
     }
 };
 
+void initialize_curve_params();
+
+class ProofVerifier {
+private:
+    bool perform_verification;
+
+    ProofVerifier(bool perform_verification) : perform_verification(perform_verification) { }
+
+public:
+    // ProofVerifier should never be copied
+    ProofVerifier(const ProofVerifier&) = delete;
+    ProofVerifier& operator=(const ProofVerifier&) = delete;
+    ProofVerifier(ProofVerifier&&);
+    ProofVerifier& operator=(ProofVerifier&&);
+
+    // Creates a verification context that strictly verifies
+    // all proofs using libsnark's API.
+    static ProofVerifier Strict();
+
+    // Creates a dummy verification context that performs
+    // no verification, used when avoiding duplicate effort
+    // such as during reindexing.
+    static ProofVerifier Dummy();
```

Comment:
Given how they are instantiated, I could see `None` being suitable for this goal. Then we'd have:

- `libzcash::ProofVerifier::Strict`
- `libzcash::ProofVerifier::None`
- `libzcash::ProofVerifier::Batch` (in future)